app/Image/GdHandler.php

```diff
@@ -0,0 +1,207 @@
+<?php
+
+namespace App\Image;
+
+use App\Image;
+use App\Logs;
+
+class GdHandler implements ImageHandlerInterface
+{
+	/**
+	 * @{inheritdoc}
+	 */
+	public function scale(
+		string $source,
+		string $destination,
+		int $newWidth,
+		int $newHeight
+	) : bool {
+		list($width, $height, $mime) = getimagesize($source);
+
+		if ($newWidth == 0) {
+			$newWidth = $newHeight * ($width / $height);
+		} else {
+			$newHeight = $newWidth / ($width / $height);
+		}
+
+		$image = imagecreatetruecolor($newWidth, $newHeight);
+		$sourceImg = $this->createImage($source, $mime);
+
+		if ($sourceImg === false) {
+			return false;
+		}
+
+		imagecopyresampled($image, $sourceImg, 0, 0, 0, 0, $newWidth, $newHeight, $width, $height);
+		imagejpeg($image, $destination, self::COMPRESSION_QUALITY);
+
+		imagedestroy($image);
+		imagedestroy($sourceImg);
+
+		return true;
+	}
+
+	/**
+	 * @{inheritdoc}
+	 */
+	public function crop(
+		string $source,
+		string $destination,
+		int $newWidth,
+		int $newHeight
+	) : bool {
+		list($width, $height, $mime) = getimagesize($source);
+
+		if ($width < $height) {
+			$newSize = $width;
+			$startWidth = 0;
+			$startHeight = $height / 2 - $width / 2;
+		} else {
+			$newSize = $height;
+			$startWidth = $width / 2 - $height / 2;
+			$startHeight = 0;
+		}
+
+		$image = imagecreatetruecolor($newWidth, $newHeight);
+		$sourceImg = $this->createImage($source, $mime);
+
+		if ($sourceImg === false) {
+			return false;
+		}
+
+		$this->fastImageCopyResampled($image, $sourceImg, 0, 0, $startWidth, $startHeight, $newWidth, $newHeight, $newSize, $newSize);
+		imagejpeg($image, $destination, self::COMPRESSION_QUALITY);
+
+		imagedestroy($image);
+		imagedestroy($sourceImg);
+
+		return true;
+	}
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function autoRotate(string $path, array $info) : array
+	{
+		$image = imagecreatefromjpeg($path);
+
+		switch ($info['orientation']) {
+			case 2:
+				imageflip($image, IMG_FLIP_HORIZONTAL);
+				break;
+
+			case 3:
+				$image = imagerotate($image, -180, 0);
+				break;
+
+			case 4:
+				imageflip($image, IMG_FLIP_VERTICAL);
+				break;
+
+			case 5:
+				$image = imagerotate($image, -90, 0);
+				imageflip($image, IMG_FLIP_HORIZONTAL);
+				break;
+
+			case 6:
+				$image = imagerotate($image, -90, 0);
+				break;
+
+			case 7:
+				$image = imagerotate($image, 90, 0);
+				imageflip($image, IMG_FLIP_HORIZONTAL);
+				break;
+
+			case 8:
+				$image = imagerotate($image, 90, 0);
+				break;
+		}
+
+		imagejpeg($image, $path, self::COMPRESSION_QUALITY);
```

Comment:
This is wrong.
With this function we are also modifying the "originals".
It should be 100.
Cf here:
https://github.com/LycheeOrg/Lychee-Laravel/pull/59/files#diff-2407954faff3ae8fbd036080db62e09fL210

Reply:
Good catch! I'll get that fixed.